back-end/yelpCrawler.py

```diff
@@ -0,0 +1,202 @@
+            businessList.append(response)
+    except HTTPError as error:
+        sys.exit(
+            'Encountered HTTP error {0} on {1}:\n {2}\nAbort program.'.format(
+                error.code,
+                error.url,
+                error.read(),
+            )
+        )
+
+    # Merge loadJsonFile.py in this file
+    datainfo = businessList
+
+    datalist = []
+
+    for item in datainfo:
+        data = dict()
+        try: data['name'] = item['name']
+        except KeyError: data['name'] = ''
+        try:
+            address = ", ".join(item['location']['display_address'])
+            data['address'] = address
+        except KeyError: data['address'] = ''
+        try: data['latitude'] = item['coordinates']['latitude']
+        except KeyError: data['latitude'] = ''
+        try: data['longitude'] = item['coordinates']['longitude']
+        except KeyError: data['longitude'] = ''
+        try: data['phone'] = item['display_phone']
+        except KeyError: data['phone']= ''
+        try: data['rating'] = item['rating']
+        except KeyError: data['rating'] = ''
+
+        opendata_list = []
+        for openitem in item['hours']:
+            for openitem_detail in openitem['open']:
+                opendata = dict()
+                try: opendata['day'] = openitem_detail['day']
+                except KeyError: opendata['day'] = ''
+                try: opendata['start'] = openitem_detail['start']
+                except KeyError: opendata['start'] = ''
+                try: opendata['end'] = openitem_detail['end']
+                except KeyError: opendata['end'] = ''
+                opendata_list.append(opendata)
+        data['openinfo'] = opendata_list
+
+        photourl = []
+        try:
+            for photoitem in item['photos']:
+                photourl.append(photoitem)
+        except KeyError: data['photourl'] = ''
+
+        data['photourl'] = photourl
+
+        # search more info with beautifulsoup and yelp_url
+        url_text = requests.get(item['url']).text
+
+        # Beautiful : Translate text to html
+        soup = BeautifulSoup(url_text, "lxml")
+
+        try:
+            website = soup.select(".biz-website > a")[0].string
+            data['website'] = website
+        except Exception as ex:
+            data['website'] = ""
+
+        moreinfo_data = dict()
+
+        moreinfo = soup.select(".short-def-list > dl")
+        for info in moreinfo:
+
+            attr_name, attr_content = "", ""
+            for dataidx in info.children:
+                if dataidx.name == "dt": #tag_name == dt
+                    attr_name = dataidx.string.strip() # strip function removes in bracket.
+                if dataidx.name == "dd": 
+                    attr_content = dataidx.string.strip()
+            moreinfo_data[attr_name] = attr_content
+
+        data['attributes'] = moreinfo_data
+        if data['attributes'].get('Wi-Fi')==wifi and data['attributes'].get('Parking') :
```

Comment:
Any condition doesn't need for Parking?

Reply:
you said to me, if parking key is not exist, there is no parking area. So I did!